source/val/instruction.h

```diff
@@ -74,13 +78,18 @@ class Instruction {
   /// Provides direct access to the stored C instruction object.
   const spv_parsed_instruction_t& c_inst() const { return inst_; }
 
+  /// Provides direct access to instructions spv_ext_inst_type_t object.
+  const spv_ext_inst_type_t& ext_inst_type() const {
+    return inst_.ext_inst_type;
+  }
+
   // Casts the words belonging to the operand under |index| to |T| and returns.
   template <typename T>
   T GetOperandAs(size_t index) const {
-    const spv_parsed_operand_t& operand = operands_.at(index);
-    assert(operand.num_words * 4 >= sizeof(T));
-    assert(operand.offset + operand.num_words <= inst_.num_words);
-    return *reinterpret_cast<const T*>(&words_[operand.offset]);
+    const spv_parsed_operand_t& op = operands_.at(index);
```

Comment:
I prefer the full word " operand" since "op" can often be confused for "opcode" especially since every SPIR-V instruction starts with "Op" as a prefix.  Having the full word here aids speed reading the code.

Reply:
Renamed to the_operand due to the operand() method taking the name.

Reply:
Even better, o, given how short it is.